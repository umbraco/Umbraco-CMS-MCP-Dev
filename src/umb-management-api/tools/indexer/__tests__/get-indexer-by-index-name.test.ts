import GetIndexerByIndexNameTool from "../get/get-indexer-by-index-name.js";
import { UmbracoManagementClient } from "@umb-management-client";
import { jest } from "@jest/globals";

const TEST_INDEX_NAME = "ExternalIndex";

describe("get-indexer-by-index-name", () => {
  let originalConsoleError: typeof console.error;
  let originalGetClient: typeof UmbracoManagementClient.getClient;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
    originalGetClient = UmbracoManagementClient.getClient;
  });

  afterEach(() => {
    console.error = originalConsoleError;
    UmbracoManagementClient.getClient = originalGetClient;
  });

  it("should get index by name", async () => {
    const result = await GetIndexerByIndexNameTool().handler(
      { indexName: TEST_INDEX_NAME },
      { signal: new AbortController().signal }
    );
    // Verify the handler response using snapshot
    expect(result).toMatchSnapshot();
  });

});