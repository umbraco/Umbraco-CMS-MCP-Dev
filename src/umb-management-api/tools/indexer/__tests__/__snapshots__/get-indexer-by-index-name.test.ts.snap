// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-indexer-by-index-name should get index by name 1`] = `
{
  "content": [
    {
      "text": "{"name":"ExternalIndex","healthStatus":{"status":"Healthy","message":null},"canRebuild":true,"searcherName":"ExternalSearcher","documentCount":62,"fieldCount":50,"providerProperties":{"CommitCount":0,"DefaultAnalyzer":"StandardAnalyzer","LuceneDirectory":"NRTCachingDirectory","LuceneIndexFolder":"/niofsdirectory@/users/philw/projects/umbraco-mcp/infrastructure/test-umbraco/mcptestsite/umbraco/data/temp/examineindexes/externalindex lockfactory=noprefixsimplefslockfactory@/users/philw/projects/umbraco-mcp/infrastructure/test-umbraco/mcptestsite/umbraco/data/temp/examineindexes/externalindex","DirectoryFactory":"Umbraco.Cms.Infrastructure.Examine.ConfigurationEnabledDirectoryFactory","EnableDefaultEventHandler":true,"PublishedValuesOnly":true,"SupportProtectedContent":false}}",
      "type": "text",
    },
  ],
}
`;
